cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests related to claiming assets trapped during XCM execution.
+
+use crate::imports::*;
+use emulated_integration_tests_common::{
+	accounts::{ALICE, BOB},
+	impls::AccountId32,
+};
+
+use frame_support::{assert_ok, sp_runtime::traits::Dispatchable};
+use xcm_executor::traits::ConvertLocation;
+use crate::imports::bhw_xcm_config::{LocationToAccountId, UniversalLocation};
+
+#[test]
+fn test_set_asset_claimer_within_a_chain() {
+	let (alice_account, alice_location) = account_and_location(ALICE);
+	let (bob_account, bob_location) = account_and_location(BOB);
+
+	let amount_to_send = 16_000_000_000_000;
+	let assets: Assets = (Parent, amount_to_send).into();
+
+	fund_account(&alice_account, amount_to_send * 2);
+	assert_eq!(query_balance(&alice_account, &RelayLocation::get()), amount_to_send * 2);
+
+	let test_args = TestContext {
+		sender: alice_account.clone(),
+		receiver: bob_account.clone(),
+		args: TestArgs::new_para(
+			bob_location.clone(),
+			bob_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), transfer_assets);
+
+	let alice_assets_after = query_balance(&alice_account, &RelayLocation::get());
+	assert_eq!(alice_assets_after, amount_to_send);
+
+	let test_args = TestContext {
+		sender: bob_account.clone(),
+		receiver: alice_account.clone(),
+		args: TestArgs::new_para(
+			alice_location.clone(),
+			alice_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), claim_assets);
+
+	let bob_assets_after = query_balance(&bob_account, &RelayLocation::get());
+	assert_eq!(bob_assets_after, amount_to_send);
+}
+
+fn account_and_location(account: &str) -> (AccountId32, Location) {
+	let account_id = PenpalA::account_id_of(account);
+	let account_clone = account_id.clone();
+	let location: Location =
+		[Junction::AccountId32 { network: Some(Westend), id: account_id.into() }].into();
+	(account_clone, location)
+}
+
+fn fund_account(account: &AccountId, amount: u128) {
+	let asset_owner = PenpalAssetOwner::get();
+	PenpalA::mint_foreign_asset(
+		<PenpalA as Chain>::RuntimeOrigin::signed(asset_owner),
+		Location::parent(),
+		account.clone(),
+		amount,
+	);
+}
```

Comment:
called in a single place, might just as well drop the helper fn

if you do keep it, rename to `penpal_a_fund_dot_account` to be clear what it does and where

Reply:
dropped